test/scripts/e2e_subs/keyreg-teal-test.sh

```diff
@@ -39,8 +39,8 @@ cat<<EOF|python - > ${TEMPDIR}/pbound
 print(((${ROUND} // ${PERIOD}) * ${PERIOD}) + ${PERIOD})
 EOF
 PBOUND=$(cat ${TEMPDIR}/pbound)
-while [ ${ROUND} != ${PBOUND} ]; do
-    goal node wait
+while [ ${ROUND} -lt ${PBOUND} ]; do
+    goal node wait --waittime 30
     ROUND=$(goal node status | grep 'Last committed block:'|awk '{ print $4 }')
 done
```

Comment:
this is a very nice bug that you've fixed. it should have it's own PR.

Reply:
@tsachiherman, Moved this change to a separate PR.

Reply:
moved to a separate PR.